### PR TITLE
Purge kubernetes.io/cluster-service from all addons

### PR DESCRIPTION
**Pull request by ialidzhikov**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
`kubernetes.io/cluster-service` is deprecated by kube-addon-manager. The label is used by kubectl cluster-info only for kind: Services resources.
The PR removes the label usage from all addons for non `kind: Service` resources.
Ref #72757.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: cluster/addons/storage-class/vsphere/default.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ metadata:
   annotations:
     storageclass.kubernetes.io/is-default-class: "true"
   labels:
-    kubernetes.io/cluster-service: "true"
+    addonmanager.kubernetes.io/mode: Reconcile
```
Comment (neolit123) [inline cluster/addons/storage-class/vsphere/default.yaml]: i think `addonmanager.kubernetes.io/mode: EnsureExists` is more appropriate for storage classes.

**Review by neolit123 — COMMENTED** [review]
thanks for this cleanup @ialidzhikov 
/ok-to-test
/kind cleanup
/priority backlog

/assign @mikedanese

File: cluster/addons/storage-class/vsphere/default.yaml (hunk 1 above)
Comment (ialidzhikov) [reply cluster/addons/storage-class/vsphere/default.yaml]: Yes, you are right. Thank you for the spot.

**Review by neolit123 — COMMENTED** [review]
/lgtm

**Comment by ialidzhikov** [comment]
Resolved the conflicts.

**Comment by ialidzhikov** [comment]
I rebase the PR now after a very long period of inactivity.

**Pull request closed without merging**